### DISCUSSION TITLE
How can I get the same useful width on different monitors?

### DISCUSSION THREAD

**Discussion by AndrewKvalheim** [paperwm/PaperWM#938] [Q&A] [answered]
I set the useful width to `1500px` and expected this to create windows of the same width on both the laptop screen and external monitor, e.g. a 100-character wide terminal—

> <p align="center"><img src="https://github.com/user-attachments/assets/74ba305d-9faf-4d18-b940-9bd14938b946" width="600" alt="external monitor with 100-character wide terminal" />  
> <img src="https://github.com/user-attachments/assets/5f30a5f5-c153-4440-9642-9f7a6fcb8a93" width="300" alt="laptop with 100-character wide terminal" /></p>

—but what actually happens is the window on the external monitor is made much wider:

> <p align="center"><img src="https://github.com/user-attachments/assets/8d30ee48-d131-4419-a3d1-07d705bff08b" width="600" alt="external monitor with 214-character wide terminal" />  
> <img src="https://github.com/user-attachments/assets/5f30a5f5-c153-4440-9642-9f7a6fcb8a93" width="300" alt="laptop with 100-character wide terminal" /></p>

How can I make the useful width consistent across monitors?

**Comment by jtaala** [comment]
What resolutions are you running?  e.g. what is your external monitor resolution vs laptop screen?

Also, post take a screenshot of useful widths, e.g.

![image](https://github.com/user-attachments/assets/dfef2371-644e-4611-9526-7412f33b94da)

**Reply by AndrewKvalheim** [reply]
```console
$ xrandr --current | grep '^\S'
Screen 0: minimum 16 x 16, current 4575 x 4320, maximum 32767 x 32767
DP-2 connected 3840x2160+0+0 (normal left inverted right x axis y axis) 600mm x 340mm
eDP-1 connected primary 3840x2160+735+2160 (normal left inverted right x axis y axis) 310mm x 170mm
$ dconf read '/org/gnome/shell/extensions/paperwm/cycle-width-steps'
[1500.0, 1892.0, 2284.0]
```

<details><summary>Screenshot</summary>

<img src="https://github.com/user-attachments/assets/254c273a-cd81-42c1-961a-6eb71d9cdce4" width="626" alt="Useful window widths: 1500px; 1892px; 2284px" />

</details>

**Comment by jtaala** [comment]
Cheers, so, interestingly, both the 1500px images above are correct.  E.g., if I take your screenshots and resize them to be the same width (e.g. those images are different sizes, so if you normalise then horizontally you get this:

![image](https://github.com/user-attachments/assets/dabe861e-89e4-4d13-8116-b50014e797eb)

So the sizes (in proportion to each monitor) is correct, e.g. the terminals are 1500px wide on each monitor.

It looks though, like the dpi (scaling) is different for each monitor.  What scaling are you using for each monitor?

**Reply by AndrewKvalheim** [reply]
The depiction in my original post and the xrandr output are accurate; eDP-1 is half the physical size of DP-2 with a scaling factor of 2.

**Comment by jtaala** [comment]
Just to close this off, this isn't supported on PaperWM.  Useful widths are always converted to actual pixel sizes.

It's conceivable the width calculation could be changed to take into account scaling factors for a monitor.  Would accept a PR to if anyone wants to implement incorporating monitor scaling into width calcs.

**Comment by AndrewKvalheim** [comment, marked as answer]
Enabling Mutter’s `scale-monitor-framebuffer` causes this to work as expected:

> <img src="https://github.com/user-attachments/assets/634ab6cd-d7a5-493a-8b6f-3c408a568f83" width="600" alt="combined screenshot of eDP-1 and DP-2 when using scale-monitor-framebuffer" />

**Reply by jtaala** [reply]
Nice find @AndrewKvalheim!

PaperWM has the ability to enable various gnome settings when it's enabled - we could add this to PaperWM settings as an enable/disable option (although I guess it's easy enough for people to enable via terminal etc.).
